packages/python/plotly/_plotly_utils/basevalidators.py

```diff
@@ -50,6 +50,83 @@ def to_scalar_or_list(v):
         return v
 
 
+plotlyjsShortTypes = {
+    "int8": "i1",
+    "uint8": "u1",
+    "int16": "i2",
+    "uint16": "u2",
+    "int32": "i4",
+    "uint32": "u4",
+    "float32": "f4",
+    "float64": "f8",
+}
+
+int8min = -128
+int8max = 127
+int16min = -32768
+int16max = 32767
+int32min = -2147483648
+int32max = 2147483647
+
+uint8max = 255
+uint16max = 65535
+uint32max = 4294967295
+
+
+def to_typed_array_spec(v):
+    """
+    Convert numpy array to plotly.js typed array spec
+    If not possible return the original value
+    """
+    v = copy_to_readonly_numpy_array(v)
+
+    np = get_module("numpy", should_load=False)
```

Comment:
Could you give some context on why this is being used instead of an import at the top?

Reply:
Presumably we're doing this because `numpy` isn't a required dependency of `plotly` (nor of `dash` for that matter) - though most people will get it when they install `pandas`. Which begs the question, what _does_ happen if you don't have `numpy` installed? Obviously then nothing is an `np.ndarray` but looks like we need to account for that possibility, as `np` will be `None` then so this would throw an error, unless we change the next line to something like
```
if np is None or not isinstance(v, np.ndarray):
```

Reply:
Thanks @alexcjohnson for the note.
`get_module("numpy", should_load=False)` is used inside other functions in this file.
So that's OK to keep it this way.

Reply:
We don’t have any tests though that DON’T have numpy available, since it’s in the CI requirements. So likely all these uses are broken for anyone who doesn’t install it, and we should protect them all. Ideally we run a few tests in an env that doesn’t install numpy at all.

Reply:
Either that or we make numpy a hard requirement, which isn’t totally out of the question, a large majority of our users surely do install it, but I think it’s better if we can keep it optional.

Reply:
Good point @alexcjohnson.
@marthacryan Could you open a separate PR from master and add a CircleCI job here 
https://github.com/plotly/plotly.py/blob/bd1dd66ed555feda1f8d514b87ddf234970a6a5a/.circleci/config.yml#L587 to test the case of numpy not being installed?
Once merged to the master I would update this branch.